src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -414,7 +419,7 @@ private void ProcessParameter(MshParameter p, PSObject inputObject, List<PSNoteP
                 }
             }
 
-            if (expressionResults.Count == 0)
+            if (expressionResults.Count == 0 && !ex.HasWildCardCharacters)
             {
                 //Commented out for bug 1107600
```

Comment:
> I think @PowerShellTeam needs to mark this moment in documentation: "make more comments in your code".

I absolutely agree! I will update [coding-conventions](https://github.com/PowerShell/PowerShell/blob/master/docs/dev-process/coding-guidelines.md#coding-conventions) to include this.